web/source/text/processor.ts

```diff
@@ -519,6 +519,12 @@ namespace com.keyman.text {
         keyman.uiManager.setActivatingUI(false);	// I2498 - KeymanWeb OSK does not accept clicks in FF when using automatic UI
       }
 
+      // Special case for embedded to pass K_TAB back to device to process
+      if(keyman.isEmbedded && (keyEvent.Lcode == Codes.keyCodes["K_TAB"] ||
+          keyEvent.Lcode == Codes.keyCodes["K_TABBACK"] || keyEvent.Lcode == Codes.keyCodes["K_TABFWD"])) {
```

Comment:
I think this would fit better inserted near [line 159 of kmwembedded.ts](https://github.com/keymanapp/keyman/pull/2546/files/cbc243e30cfd856c7aec3d444133a3c2a0876409..90c441857d921d40075e97e1b53ea408a3a81216#diff-0b102a616ca81cfcf2c978ee673383d3R159) instead, modifying the return value there.

Reply:
from what I can tell, that only returns strings, so I didn't have a way to return `false`

Reply:
Right, it's about JS event processing management.  My mistake.

Reply:
I'm curious why this wouldn't eventually fall through to the embedded app anyway? What if we had a keyboard that wanted to do processing on TAB?